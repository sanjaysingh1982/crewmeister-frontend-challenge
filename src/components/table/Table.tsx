import { MDBDataTable } from 'mdbreact';
import PropTypes from 'prop-types';

const Table = ({ events }: any) => {
    const data = {
        columns: [
            {
                label: 'Title',
                field: 'title',
                sort: 'asc',
            },
            {
                label: 'Type',
                field: 'type',
                sort: 'asc',
            },
            {
                label: 'Start Date',
                field: 'startDate',
                sort: 'asc',
            },
            {
                label: 'End Date',
                field: 'endDate',
                sort: 'asc',
            },
            {
                label: 'Member Note',
                field: 'memberNote',
                sort: 'asc',
            },
            {
                label: 'Status',
                field: 'status',
                sort: 'asc',
            },
            {
                label: 'Admitter Note',
                field: 'admitterNote',
                sort: 'asc',
            }
        ],
        rows: events
    };

    return (
        <MDBDataTable
            striped
            bordered
            small
            noBottomColumns={true}
            data={data}
        />
    );
};

Table.propTypes = {
    events: PropTypes.array
};

Table.defualtProps = {
    events: []
};

export default Table;