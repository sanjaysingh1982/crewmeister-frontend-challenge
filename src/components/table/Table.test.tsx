import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Table from './Table';


const setup = (props = {}, state = null) => {
    const defaultProps = {
        events: [{
            'admitterId': null,
            'admitterNote': '',
            'confirmedAt': '2016-12-12T18:03:55.000+01:00',
            'createdAt': '2016-12-12T14:17:01.000+01:00',
            'crewId': 352,
            'endDate': '2017-01-13',
            'id': 2351,
            'memberNote': '',
            'rejectedAt': null,
            'startDate': '2017-01-13',
            'type': 'sickness',
            'userId': 2664
        },
            {
                'admitterId': null,
                'admitterNote': '',
                'confirmedAt': '2017-01-03T17:39:50.000+01:00',
                'createdAt': '2017-01-03T17:36:52.000+01:00',
                'crewId': 352,
                'endDate': '2017-01-05',
                'id': 2521,
                'memberNote': 'ganzer tag',
                'rejectedAt': null,
                'startDate': '2017-01-05',
                'type': 'vacation',
                'userId': 2664
            }
        ]
    };
    const setUpProps = {...defaultProps, ...props};
    const wrapper = shallow(<Table {...setUpProps}/>);
    const tree = () => renderer.create(<Table {...setUpProps}/>).toJSON();
    if (state) wrapper.setState(state);
    return {wrapper, props: setUpProps, tree};
};

describe('Table component', () => {
    it('should match with snapshot', () => {
        const {tree} = setup();
        expect(tree()).toMatchSnapshot();
    });
});