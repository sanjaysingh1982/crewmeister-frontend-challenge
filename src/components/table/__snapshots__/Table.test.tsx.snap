// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table component should match with snapshot 1`] = `
<div
  className="dataTables_wrapper dt-bootstrap4"
  data-test="datatable"
>
  <div
    className="row"
  >
    <div
      className="col-sm-12 col-md-6"
      data-test="datatable-entries"
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "justifyContent": false,
        }
      }
    >
      <div
        className="dataTables_length bs-select"
        data-test="datatable-select"
      >
        <label>
          Show entries
          <select
            className="custom-select custom-select-sm form-control form-control-sm"
            onChange={[Function]}
            style={
              Object {
                "marginLeft": ".5rem",
              }
            }
            value={10}
          >
            <option
              value={10}
            >
              10
            </option>
            <option
              value={20}
            >
              20
            </option>
            <option
              value={50}
            >
              50
            </option>
            <option
              value={100}
            >
              100
            </option>
          </select>
        </label>
      </div>
    </div>
    <div
      className="col-sm-12 col-md-6"
      data-test="datatable-search"
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "justifyContent": "flex-end",
        }
      }
    >
      <div
        className="mdb-datatable-filter flex-row"
        data-test="datatable-input"
      >
        <input
          aria-label="Search"
          className="form-control form-control-sm ml-0 my-1"
          onChange={[Function]}
          placeholder="Search"
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-sm-12"
      data-test="datatable-table"
    >
      <div
        className=""
        data-test="table"
        style={
          Object {
            "maxHeight": undefined,
          }
        }
      >
        <table
          className="table table-bordered table-sm table-striped dataTable"
          entries={10}
        >
          <thead
            data-test="datatable-head"
          >
            <tr>
              <th
                className="sorting"
                onClick={[Function]}
              >
                Title
              </th>
              <th
                className="sorting"
                onClick={[Function]}
              >
                Type
              </th>
              <th
                className="sorting"
                onClick={[Function]}
              >
                Start Date
              </th>
              <th
                className="sorting"
                onClick={[Function]}
              >
                End Date
              </th>
              <th
                className="sorting"
                onClick={[Function]}
              >
                Member Note
              </th>
              <th
                className="sorting"
                onClick={[Function]}
              >
                Status
              </th>
              <th
                className="sorting"
                onClick={[Function]}
              >
                Admitter Note
              </th>
            </tr>
          </thead>
          <tbody
            data-test="table-body"
          >
            <tr>
              <td />
              <td>
                sickness
              </td>
              <td>
                2017-01-13
              </td>
              <td>
                2017-01-13
              </td>
              <td />
              <td />
              <td />
            </tr>
            <tr>
              <td />
              <td>
                vacation
              </td>
              <td>
                2017-01-05
              </td>
              <td>
                2017-01-05
              </td>
              <td>
                ganzer tag
              </td>
              <td />
              <td />
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-sm-12 col-md-5"
      data-test="datatable-info"
    >
      <div
        aria-live="polite"
        className="dataTables_info"
        role="status"
      >
        Showing 1 to 2 of 2 entries
      </div>
    </div>
    <div
      className="col-sm-12 col-md-7"
      data-test="datatable-pagination"
    >
      <div
        className="dataTables_paginate"
      >
        <ul
          className="pagination"
          data-test="pagination"
        >
          <li
            className="disabled page-item"
            data-test="page-item"
          >
            <a
              aria-label="Previous"
              className="page-link page-link"
              data-test="page-link"
              onClick={[Function]}
            >
              <span>
                Previous
              </span>
            </a>
          </li>
          <li
            className="active page-item"
            data-test="page-item"
          >
            <a
              className="page-link page-link"
              data-test="page-link"
              onClick={[Function]}
            >
              1
              <span
                className="sr-only"
              >
                (current)
              </span>
            </a>
          </li>
          <li
            className="disabled page-item"
            data-test="page-item"
          >
            <a
              aria-label="Next"
              className="page-link page-link"
              data-test="page-link"
              onClick={[Function]}
            >
              <span>
                Next
              </span>
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
